Regression: `FromIterator<TokenStream>` is not implemented for `TokenStream`

Servo doesn’t build in today’s Nightly:

```rust
error[E0277]: the trait bound `proc_macro::TokenStream: std::iter::FromIterator<proc_macro::TokenStream>` is not satisfied
  --> components/dom_struct/lib.rs:26:53
   |
26 |     iter::once(attributes).chain(iter::once(input)).collect()
   |                                                     ^^^^^^^ a collection of type `proc_macro::TokenStream` cannot be built from an iterator over elements of type `proc_macro::TokenStream`
   |
   = help: the trait `std::iter::FromIterator<proc_macro::TokenStream>` is not implemented for `proc_macro::TokenStream`
error: aborting due to previous error
```

Since the `TokenStream` type is stable I *think* this is a breaking change.

CC @alexcrichton for https://github.com/rust-lang/rust/pull/49597.

--- a/src/libproc_macro/lib.rs
+++ b/src/libproc_macro/lib.rs
@@ -141,9 +141,16 @@ impl From<TokenTree> for TokenStream {
 #[unstable(feature = "proc_macro", issue = "38356")]
 impl iter::FromIterator<TokenTree> for TokenStream {
     fn from_iter<I: IntoIterator<Item = TokenTree>>(trees: I) -> Self {
+        trees.into_iter().map(TokenStream::from).collect()
+    }
+}
+
+#[unstable(feature = "proc_macro", issue = "38356")]
+impl iter::FromIterator<TokenStream> for TokenStream {
+    fn from_iter<I: IntoIterator<Item = TokenStream>>(streams: I) -> Self {
         let mut builder = tokenstream::TokenStreamBuilder::new();
-        for tree in trees {
-            builder.push(tree.to_internal());
+        for stream in streams {
+            builder.push(stream.0);
         }
         TokenStream(builder.build())
     }

